libqtile/utils.py

```diff
@@ -209,3 +209,38 @@ def send_notification(title, message, urgent=False, timeout=10000):
         notifier.show()
     except Exception as exception:
         logger.error(exception)
+
+
+def guess_terminal():
+    """Try to guess terminal."""
+    test_terminals = [
+            ['x-terminal-emulator', '-?'],
+            ['xterm', '-help'],
+            ['urxvt', '-help'],
+            ['st', '-v'],
+            ['xfce4-terminal', '-V'],
+            ['gnome-terminal', '-?'],
+            ['terminator', '-v'],
+            ['lxterminal', '-v'],
+            ['mate-terminal', '-h'],
+            ['roxterm', '-?'],
+            ['tilda', '--help'],
+            ['guake', '-h'],
+            ['eterm', '-h'],
+            ['sakura', '-v'],
+            ['konsole', '--help'],
+            ['hyper', '-v'],
+            ['alacritty', '-V'],
```

Comment:
I think I would revert the list so that the more "modern" and "powerful" terminals can be found first. Because if such a terminal is present, it means that the user installed it for its own use, and thus would probably prefer to see it spawn rather than something like xterm. Is it a fair assumption?

For example, I currently have roxterm, urxvt and xterm installed, and I would like them to be found in that order.

Reply:
let's try to build queue

            ['x-terminal-emulator', '-?'],
            ['roxterm', '-?'],
            ['sakura', '-v'],
            ['hyper', '-v'],
            ['alacritty', '-V'],
            ['terminator', '-v'],
            ['gnome-terminal', '-?'],
            ['konsole', '--help'],
            ['xfce4-terminal', '-V'],
            ['lxterminal', '-v'],
            ['mate-terminal', '-h'],
            ['tilda', '--help'],
            ['guake', '-h'],
            ['eterm', '-h'],
            ['st', '-v'],
            ['urxvt', '-help'],
            ['xterm', '-help'],

Reply:
I'd put tilda and guake above gnome-terminal, roxterm under terminator, and eterm under urxvt. It looks pretty good to me otherwise!

Reply:
I disagree about guake and tilda - they are not real terminals :D and I don't know how they should look if I press new terminal twice. I don't think that we will find ideal way to represent all people. The main reason for this is to have terminal then you start. I think list should be as it is and maybe from practice we will see what is not in right place. In my case I will not use this utility myself and after installing os I will install my loved terminal and it will be alone (st), other just for testing (xterm because of Qtile dev and Alacritty because I havent removed it yet)

Reply:
I agree, but then let's remove guake and tilda? I don't think anyone has only one of these two installed anyway.